backend/pkg/kubeconfig/kubeconfig.go

```diff
@@ -316,299 +335,381 @@ func loadContextsFromData(data []byte, source int) ([]Context, []error) {
-func createKubeContext(contextData map[interface{}]interface{}) *api.Context {
-	kubeContext := &api.Context{}
+// createKubeConfig creates a kubeconfig from the given context, cluster, and user.
+func createKubeConfig(
+	contextName string,
+	context,
+	cluster,
+	user map[interface{}]interface{},
+) map[string]interface{} {
+	kubeconfig := make(map[string]interface{})
+	kubeconfig["contexts"] = []interface{}{context}
+	kubeconfig["clusters"] = []interface{}{cluster}
+	kubeconfig["users"] = []interface{}{user}
+	kubeconfig["apiVersion"] = "v1"
+	kubeconfig["kind"] = "Config"
+	kubeconfig["current-context"] = contextName
 
-	if cluster, ok := contextData["cluster"].(string); ok {
-		kubeContext.Cluster = cluster
+	return kubeconfig
+}
+
+// convertToContext converts the config to a context.
+func convertToContext(contextName string, clientConfig *api.Config, source int) (Context, error) {
+	context, exists := clientConfig.Contexts[contextName]
+	if !exists {
+		return Context{}, ContextError{
+			ContextName: contextName,
+			Reason:      "context not found in loaded config",
+		}
 	}
 
-	if namespace, ok := contextData["namespace"].(string); ok {
-		kubeContext.Namespace = namespace
+	cluster, exists := clientConfig.Clusters[context.Cluster]
+	if !exists {
+		return Context{}, ClusterError{
+			ClusterName: context.Cluster,
+			Reason:      "cluster not found in loaded config",
+		}
 	}
 
-	if user, ok := contextData["user"].(string); ok {
-		kubeContext.AuthInfo = user
+	authInfo, exists := clientConfig.AuthInfos[context.AuthInfo]
+	if !exists {
+		return Context{}, UserError{
+			UserName: context.AuthInfo,
+			Reason:   "user not found in loaded config",
+		}
 	}
 
-	// Handle extensions if present
-	if extensions, ok := contextData["extensions"].(map[interface{}]interface{}); ok {
-		kubeContext.Extensions = make(map[string]k8sruntime.Object)
+	// Make contextName DNS friendly.
+	contextName = makeDNSFriendly(contextName)
 
-		for k, v := range extensions {
-			if key, ok := k.(string); ok {
-				if obj, ok := v.(k8sruntime.Object); ok {
-					kubeContext.Extensions[key] = obj
-				} else {
-					// If the extension is not already a runtime.Object,
-					// wrap it in a CustomObject
-					kubeContext.Extensions[key] = &CustomObject{
-						TypeMeta: metav1.TypeMeta{
-							Kind:       "CustomObject",
-							APIVersion: "v1",
-						},
-						ObjectMeta: metav1.ObjectMeta{
-							Name: key,
-						},
-						CustomName: key,
-					}
-				}
-			}
-		}
+	newContext := Context{
+		Name:        contextName,
+		KubeContext: context,
+		Cluster:     cluster,
+		AuthInfo:    authInfo,
+		Source:      source,
+	}
+
+	err := newContext.SetupProxy()
+	if err != nil {
+		return Context{}, ContextError{ContextName: contextName, Reason: fmt.Sprintf("couldn't setup proxy: %v", err)}
 	}
 
-	return kubeContext
+	return newContext, nil
 }
 
 // LoadContextsFromAPIConfig loads contexts from the given api.Config.
```

Comment:
The `LoadContextsFromAPIConfig` function is not used anywhere in the code, remove it if it is not used. If it is replaced by another function then make sure that it also supports the `skipProxySetup` functionality.